qiskit_experiments/database_service/json.py

```diff
@@ -29,13 +32,25 @@ def default(self, obj: Any) -> Any:  # pylint: disable=arguments-differ
         if isinstance(obj, complex):
             return {"__type__": "complex", "__value__": [obj.real, obj.imag]}
         if isinstance(obj, FitVal):
-            return {"__type__": obj.__class__.__name__, "args": (obj.value, obj.stderr, obj.unit)}
-        if callable(obj):
-            return {"__type__": "callable", "__value__": obj.__name__}
+            return {"__type__": type(obj).__name__, "args": (obj.value, obj.stderr, obj.unit)}
+        if isinstance(obj, (Operator, Choi)):
+            return {
+                "__type__": type(obj).__name__,
+                "args": (obj.data,),
+                "kwargs": {"input_dims": obj.input_dims(), "output_dims": obj.output_dims()},
+            }
+        if isinstance(obj, (Statevector, DensityMatrix)):
+            return {
+                "__type__": type(obj).__name__,
+                "args": (obj.data,),
+                "kwargs": {"dims": obj.dims()},
+            }
+        if isinstance(obj, FunctionType):
+            return {"__type__": "function", "__value__": obj.__name__}
```

Comment:
@jyu00 added this originally so i'm not sure how to recover it. My change was to switch the type check from `Callable` to `FunctionType` since i had issues with classes with call methods like `quantum_info.Operator`